VideoClassification/readers.py

```diff
@@ -1,347 +1,200 @@
-      uniq_start_times, seg_idxs = tf.unique(start_times,
-                                             out_idx=tf.dtypes.int64)
-      # TODO(zhengxu): Ensure the segment_sizes are all same.
-      segment_size = self.segment_size
-      # Range gather matrix, e.g., [[0,1,2],[1,2,3]] for segment_size == 3.
-      range_mtx = tf.expand_dims(uniq_start_times, axis=-1) + tf.expand_dims(
-          tf.range(0, segment_size, dtype=tf.int64), axis=0)
-      # Shape: [num_segment, segment_size, feature_dim].
-      batch_video_matrix = tf.gather_nd(video_matrix,
-                                        tf.expand_dims(range_mtx, axis=-1))
-      num_segment = tf.shape(batch_video_matrix)[0]
-      batch_video_ids = tf.reshape(tf.tile([contexts["id"]], [num_segment]),
-                                   (num_segment,))
-      batch_frames = tf.reshape(tf.tile([segment_size], [num_segment]),
-                                (num_segment,))
-
-      # For segment labels, all labels are not exhausively rated. So we only
-      # evaluate the rated labels.
-
-      # Label indices for each segment, shape: [num_segment, 2].
-      label_indices = tf.stack([seg_idxs, contexts["segment_labels"].values],
-                               axis=-1)
-      label_values = contexts["segment_scores"].values
-      sparse_labels = tf.sparse.SparseTensor(label_indices, label_values,
-                                             (num_segment, self.num_classes))
-      batch_labels = tf.sparse.to_dense(sparse_labels, validate_indices=False)
-
-      sparse_label_weights = tf.sparse.SparseTensor(
-          label_indices, tf.ones_like(label_values, dtype=tf.float32),
-          (num_segment, self.num_classes))
-      batch_label_weights = tf.sparse.to_dense(sparse_label_weights,
-                                               validate_indices=False)
-    else:
-      # Process video-level labels.
-      label_indices = contexts["labels"].values
-      sparse_labels = tf.sparse.SparseTensor(
-          tf.expand_dims(label_indices, axis=-1),
-          tf.ones_like(contexts["labels"].values, dtype=tf.bool),
-          (self.num_classes,))
-      labels = tf.sparse.to_dense(sparse_labels,
-                                  default_value=False,
-                                  validate_indices=False)
-      # convert to batch format.
-      batch_video_ids = tf.expand_dims(contexts["id"], 0)
-      batch_video_matrix = tf.expand_dims(video_matrix, 0)
-      batch_labels = tf.expand_dims(labels, 0)
-      batch_frames = tf.expand_dims(num_frames, 0)
-      batch_label_weights = None
-
-    output_dict = {
-        "video_ids": batch_video_ids,
-        "video_matrix": batch_video_matrix,
-        "labels": batch_labels,
-        "num_frames": batch_frames,
-    }
-    if batch_label_weights is not None:
-      output_dict["label_weights"] = batch_label_weights
-
-    return output_dict
+	"""Truncates or pads a tensor to new_size on on a given axis.
```

Comment:
I'd suggest moving this function to a util file, instead of putting it in a file called "reader"

Reply:
done